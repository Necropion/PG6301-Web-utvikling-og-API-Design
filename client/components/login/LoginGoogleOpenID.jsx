import { useEffect, useState } from "react";

const LoginGoogleOpenID = () => {
  const [authURL, setAuthURL] = useState("");

  const google_discovery_url =
    "https://accounts.google.com/.well-known/openid-configuration";
  const google_client_id =
    "673274078987-vv9i592bvrr7fff9o7mfftnfg5kfspgu.apps.googleusercontent.com";

  const loadAuthURL = async () => {
    try {
      const response = await fetch(google_discovery_url);

      if (response.ok) {
        const discoveryDoc = await response.json();

        const params = {
          response_type: "token",
          scope: "email profile",
          client_id: google_client_id,
          redirect_uri: window.location.origin + "/callback",
        };

        setAuthURL(
          discoveryDoc.authorization_endpoint +
            "?" +
            new URLSearchParams(params),
        );
      }

      if (!response.ok) {
        const errorMsg = `Failed fetching 'discovery url': ${response.status} ${response.statusText}`;
        console.log(errorMsg);
      }
    } catch (error) {
      console.log(error);
    }
  };

  useEffect(() => {
    loadAuthURL();
  }, []);

  return (
    <a className="googleBtn" href={authURL}>
      Google
    </a>
  );
};

export default LoginGoogleOpenID;
