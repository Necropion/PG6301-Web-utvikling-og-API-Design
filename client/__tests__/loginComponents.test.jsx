import renderer from "react-test-renderer";
import LoginGoogleOpenID from "../components/login/LoginGoogleOpenID";

describe("Login components tests", () => {
  it("Renders Google login button", () => {
    const component = renderer.create(<LoginGoogleOpenID />);
    expect(component).toMatchSnapshot();
  });
});
