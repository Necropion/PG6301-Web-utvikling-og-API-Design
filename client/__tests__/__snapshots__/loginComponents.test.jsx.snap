// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login components tests Renders Google login button 1`] = `
<a
  className="googleBtn"
  href=""
>
  Google
</a>
`;
